admin-ui/src/app/shared/dialog/dialog.component.ts

```diff
@@ -124,13 +124,31 @@ export class DialogComponent implements OnInit {
         .subscribe(response => {
           // tslint:disable-next-line:no-string-literal
           this.FilterData = [...response['filterColumns']];
-          // tslint:disable-next-line:no-string-literal
+          this.FilterData.forEach(values => {
+            if(values.filtername === "locationCode")
+              this.getLocationHierarchyLevels();
+          });
           this.settingUpFilter(response['filterColumns']);
+          // tslint:disable-next-line:no-string-literal          
           resolve(true);
         });
     });
   }
 
+  getLocationHierarchyLevels() {
+    let self = this;
+    let fieldNameData = {};
+    this.dataStorageService.getLocationHierarchyLevels(this.primaryLangCode).subscribe(response => {
+      response.response.locationHierarchyLevels.forEach(function (value) {
+        if(value.hierarchyLevel == self.config.getConfig()['locationHierarchyLevel'])            
+          self.FilterData.forEach((values, index) => {
+            if(values.filtername === self.primaryLangCode)
+              self.FilterData[index].filterlabel["eng"] = values.filtername;
```

Comment:
Donot hardcode langCode ?

Reply:
The code used to bring the dynamic location code and checking the label is present or not.
This code currently not used effectively.